vscode/microsoft-kiota/src/kiotaInterop/index.ts

```diff
@@ -1,25 +1,36 @@
 import * as cp from 'child_process';
-import * as vscode from "vscode";
+import * as net from 'node:net';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import * as vscode from 'vscode';
 import * as rpc from 'vscode-jsonrpc/node';
 
 import { KiotaGenerationLanguage, KiotaPluginType } from '../types/enums';
 import { getWorkspaceJsonDirectory } from "../util";
 import { ensureKiotaIsPresent, getKiotaPath } from './kiotaInstall';
 
+const extensionSanitizerRe = /[^a-zA-Z0-9_]+/g;
 export async function connectToKiota<T>(context: vscode.ExtensionContext, callback: (connection: rpc.MessageConnection) => Promise<T | undefined>, workingDirectory: string = getWorkspaceJsonDirectory()): Promise<T | undefined> {
     const kiotaPath = getKiotaPath(context);
     await ensureKiotaIsPresent(context);
-    const childProcess = cp.spawn(kiotaPath, ["rpc"], {
+    // Use a unique pipe for this extension.
+    const re = extensionSanitizerRe;
+    const suffix = context.extension.id.replace(re, '_');
+    const pipeName = `KiotaJsonRpc-${suffix}`;
```

Comment:
*question:* Since the implementation of the connectToKiota function is going to be more generic as described in this PR: https://github.com/microsoft/kiota/pull/6172, how are we going to define the pipeNames on the fly?

Reply:
It could be a parameter passed in by the extension, no?

Reply:
I've changed the function to use a uuid instead.